yarn:install on windows would crash

joshleblanc opened the pull request:
### Summary

Windows can't implicitly run the `bin/yarn` using backticks, as it throws an `Errno::EINVAL` if you try. This change adds the ruby interpreter path to the call to let the OS know what to run the file with.

gmcgibbon commented on railties/lib/rails/tasks/yarn.rake:
```diff
@@ -10,7 +10,7 @@ namespace :yarn do
 
     yarn_flags =
-      if `"#{Rails.root}/bin/yarn" --version`.start_with?("1")
+      if `#{RbConfig.ruby} "#{Rails.root}/bin/yarn" --version`.start_with?("1")
```
Please add this prefix to the `yarn install` line as well. The install command itself is also currently broken on Windows.

gmcgibbon reviewed (commented):
It seems to raise `Errno::ENOEXEC`, but indeed this is broken.

This is the same [solution](https://github.com/rails/webpacker/blob/b7069f4618de51415a4449908346a3c0853880dc/lib/tasks/webpacker/install.rake#L10) the webpacker install script uses. 👍 

Could you address the comment and write a more detailed commit message? Thanks!

joshleblanc committed:
execute yarn executable using current ruby interpreter

This change is required to support windows. It explicitly runs the bin/yarn file using the ruby executable that's running the rake task.
